examples/fuse_examples/multimodality/image_clinical/multimodality_image_clinical.ipynb

```diff
@@ -1005,43 +371,46 @@
    "source": [
     "from fuse.utils.ndict import NDict\n",
     "import torch\n",
+    "from fuse.data import OpBase\n",
     "from fuse_examples.multimodality.image_clinical.dataset import ANATOM_SITE_INDEX, SEX_INDEX\n",
     "\n",
     "### Generate Data\n",
-    "def post_cache_processing_clinical_encoding(sample_dict: NDict) -> dict:\n",
-    "    age = sample_dict['data.input.clinical.age_approx']\n",
-    "    if age > 0 and age < 120:\n",
-    "        age = torch.tensor(age / 120.0).reshape(-1)\n",
-    "    else:\n",
-    "        age = torch.tensor(-1.0).reshape(-1)\n",
-    "    \n",
-    "    anatom_site = sample_dict['data.input.clinical.anatom_site_general']\n",
-    "    anatom_site_one_hot = torch.zeros(len(ANATOM_SITE_INDEX))\n",
-    "    if anatom_site in ANATOM_SITE_INDEX:\n",
-    "        anatom_site_one_hot[ANATOM_SITE_INDEX[anatom_site]] = 1\n",
-    "    \n",
-    "    sex = sample_dict['data.input.clinical.sex']\n",
-    "    sex_one_hot = torch.zeros(len(SEX_INDEX))\n",
-    "    if sex in SEX_INDEX:\n",
-    "        sex_one_hot[SEX_INDEX[sex]] = 1\n",
-    "    \n",
-    "    clinical_encoding = torch.cat((age, anatom_site_one_hot, sex_one_hot), dim=0)\n",
-    "    sample_dict[\"data.input.clinical.all\"] = clinical_encoding\n",
-    "\n",
-    "    return sample_dict\n",
-    "\n",
-    "\n",
-    "train_dl, valid_dl = isic_2019_dataset(size=size, reset_cache=False, post_cache_processing_func=post_cache_processing_clinical_encoding)\n",
+    "class OpClinicalEncodeing(OpBase):\n",
+    "    def __call__(self, sample_dict: NDict) -> NDict:\n",
+    "        \n",
+    "        age = sample_dict['data.input.clinical.age_approx']\n",
+    "        if age > 0 and age < 120:\n",
+    "            age = torch.tensor(age / 120.0).reshape(-1)\n",
+    "        else:\n",
+    "            age = torch.tensor(-1.0).reshape(-1)\n",
+    "        \n",
+    "        anatom_site = sample_dict['data.input.clinical.anatom_site_general']\n",
+    "        anatom_site_one_hot = torch.zeros(len(ANATOM_SITE_INDEX))\n",
+    "        if anatom_site in ANATOM_SITE_INDEX:\n",
+    "            anatom_site_one_hot[ANATOM_SITE_INDEX[anatom_site]] = 1\n",
+    "        \n",
+    "        sex = sample_dict['data.input.clinical.sex']\n",
+    "        sex_one_hot = torch.zeros(len(SEX_INDEX))\n",
+    "        if sex in SEX_INDEX:\n",
+    "            sex_one_hot[SEX_INDEX[sex]] = 1\n",
+    "        \n",
+    "        clinical_encoding = torch.cat((age, anatom_site_one_hot, sex_one_hot), dim=0)\n",
+    "        sample_dict[\"data.input.clinical.all\"] = clinical_encoding\n",
+    "\n",
+    "        return sample_dict\n",
+    "\n",
+    "append_dyn_pipeline = [(OpClinicalEncodeing(), dict())]\n",
```

Comment:
why do you add the dict() here? 
maybe also explain in the tutorial

Reply:
Done